product_taxes_group/model/product_template.py

```diff
@@ -42,13 +41,15 @@ class ProductTemplate(models.Model):
     # Overload Section
     @api.model
     def create(self, vals):
-        self.check_coherent_vals(False, vals)
-        return super(ProductTemplate, self).create(vals)
+        res = super(ProductTemplate, self).create(vals)
+        res.check_coherent_vals(vals)
+        return res
 
     @api.multi
     def write(self, vals):
-        self.check_coherent_vals([x.id for x in self], vals)
-        return super(ProductTemplate, self).write(vals)
+        super(ProductTemplate, self).write(vals)
+        self.check_coherent_vals(vals)
+        return True
```

Comment:
write function should return res with res = super(ProductTemplate, self).write(vals)